components/tidb_query/src/expr/builtin_math.rs

```diff
@@ -157,13 +156,13 @@ impl ScalarFunc {
     pub fn rand(&self, _: &mut EvalContext, _: &[Datum]) -> Result<Option<f64>> {
         let mut cus_rng = self.cus_rng.rng.borrow_mut();
         if cus_rng.is_none() {
-            let mut rand = get_rand(None);
-            let res = rand.gen::<f64>();
-            *cus_rng = Some(rand);
+            let rng = get_rng(None);
+            let res = (rng.borrow_mut()).gen::<f64>();
+            *cus_rng = Some(rng.borrow_mut().to_owned());
```

Comment:
`RefCell::new()` here

Reply:
why we need new at here? could we just use bare?

Reply:
Because I think it's not a good idea for `get_rng` to return a `RefCell` which is a callee requirement.

Reply:
I know what you mean, but I think in impl we can new refcell in there.
But in buildin I think we can just use bare, or we need also cache in buildin.

Reply:
I think we've already cached rng in builtin

https://github.com/tikv/tikv/blob/b72b92ed141e8bc8641feb8226f9d027a3458180/components/tidb_query/src/expr/mod.rs#L71

Reply:
So why we need new refcell in buildin?

Reply:
I mean that if get_rng() returns a bare Rng, we have to wrap it into RefCell here.

Reply:
I still think there is indeed to new RefCell, because * cus_rng already is XorShiftRng.